airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -2,41 +2,65 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
 import logging
+import os
+import traceback
 from io import BytesIO, IOBase
-from typing import Any, Dict, Iterable, List, Mapping, Optional
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Tuple, Union
 
+import backoff
+import dpath.util
+import requests
 from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
-from airbyte_cdk.sources.file_based.config.unstructured_format import UnstructuredFormat
+from airbyte_cdk.sources.file_based.config.unstructured_format import (
+    APIParameterConfigModel,
+    APIProcessingConfigModel,
+    LocalProcessingConfigModel,
+    UnstructuredFormat,
+)
 from airbyte_cdk.sources.file_based.exceptions import FileBasedSourceError, RecordParseError
 from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
 from airbyte_cdk.sources.file_based.file_types.file_type_parser import FileTypeParser
 from airbyte_cdk.sources.file_based.remote_file import RemoteFile
 from airbyte_cdk.sources.file_based.schema_helpers import SchemaType
-from unstructured.documents.elements import Formula, ListItem, Title
-from unstructured.file_utils.filetype import STR_TO_FILETYPE, FileType, detect_filetype
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from unstructured.file_utils.filetype import FILETYPE_TO_MIMETYPE, STR_TO_FILETYPE, FileType, detect_filetype
 
 unstructured_partition_pdf = None
 unstructured_partition_docx = None
 unstructured_partition_pptx = None
-unstructured_optional_decode = None
+
+
+def optional_decode(contents: Union[str, bytes]) -> str:
+    if isinstance(contents, bytes):
+        return contents.decode("utf-8")
+    return contents
 
 
 def _import_unstructured() -> None:
     """Dynamically imported as needed, due to slow import speed."""
     global unstructured_partition_pdf
     global unstructured_partition_docx
     global unstructured_partition_pptx
-    global unstructured_optional_decode
     from unstructured.partition.docx import partition_docx
-    from unstructured.partition.md import optional_decode
     from unstructured.partition.pdf import partition_pdf
     from unstructured.partition.pptx import partition_pptx
 
     # separate global variables to properly propagate typing
     unstructured_partition_pdf = partition_pdf
     unstructured_partition_docx = partition_docx
     unstructured_partition_pptx = partition_pptx
-    unstructured_optional_decode = optional_decode
+
+
+def user_error(e: Exception) -> bool:
+    """
+    Return True if this exception is caused by user error, False otherwise.
+    """
+    if not isinstance(e, requests.exceptions.RequestException):
+        return False
+    return bool(e.response and 400 <= e.response.status_code < 500)
+
+
+CLOUD_DEPLOYMENT_MODE = "cloud"
```

Comment:
nit: this value is defined in `entrypoint.py`. It might be worth moving it to a separate file so it can be shared